src/main/java/org/terasology/smithing/component/CharcoalPitComponent.java

```diff
@@ -19,16 +19,28 @@
 import org.terasology.network.Replicate;
 
 public class CharcoalPitComponent implements Component {
+
+    /** The time in milliseconds in-game when the charcoal pit begins to burn **/
```

Comment:
You don't need two *s at the end - javadoc goes `/** ... */` :-)

Also, slight potential tweak: would it be better to say "when the charcoal pit _began_ to burn" ? Is it expected to have a value, like when the burning began and the component was added? Or is the component always present and this field holds the last time or something else? (uninitialized / -1?)